lib/PuppeteerSharp/NavigatorWatcher.cs

```diff
@@ -37,7 +37,7 @@ public NavigatorWatcher(FrameManager frameManager, Frame mainFrame, int timeout,
 
             _expectedLifecycle = waitUntil.Select(w =>
             {
-                var protocolEvent = _puppeteerToProtocolLifecycle.GetValueOrDefault(w);
+                var protocolEvent = DictionaryExtensions.GetValueOrDefault(_puppeteerToProtocolLifecycle, w);

```

Comment:
why can't the method be called as an extension?

Reply:
I think compiler is right, and this will persist in the following dotnet versions. 
What difference of this extension method between `System.Collections.Generic.CollectionExtensions.GetValueOrDefault`?

Reply:
@kblok

Reply:
The problem is `GetValueOrDefault` exists in different namespaces `System.Collections.Generic` and `DictionaryExtensions.GetValueOrDefault`. Both namespaces are used, because of that compile error with ambiguous calls. 
I dont understand the purpose of DictionaryExtensions, at first sight it does the same as standart library.

Reply:
We can't use `System.Collections.Generic.CollectionExtensions`
Puppeteer-Sharp project must be a NetStandard Library.

Reply:
So it's proper solution or it should be done other way like @Meir017 wrote? 
I get ambiguous calls on every version of dotnet core (2.0 and 2.1).

Reply:
The thing is that PuppetterSharp is NetStandard, not NetCore. Can you try removing the Target frameworks you added?

Reply:
Ok, I reverted these changes. I'm able to build the project with mono msbuild but I did succeed with dotnet core. 

dotnet core has built-it msbuild which can be run with `dotnet msbuild`, by default it's required .NETFramework,Version=v4.7.1 for this project, but this framework only for Windows. I tried run it like this:
`dotnet msbuild /p:TargetFrameworkIdentifier=.NETCoreApp /p:TargetFrameworkVersion=2.1`, but actully it's the same as `dotnet build -f dotnet netcoreapp2.1` and it's required the appropriate target to be defined in csproj.

Reply:
@HarpyWar on Linux we only need to run the tests. We would package the app on Windows. This should work on Linux:
```
dotnet test -f netcoreapp2.0
```

`dotnet test` will build all the dependencies.